src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -0,0 +1,426 @@
+        will be available.
+        :param uninjectedData: same data as it's returned from the uninjectedFiles
+        :return: a dictionary of successfully inserted blocks and their correspondent location
+        """
+        logging.info("Preparing to insert blocks into Rucio...")
+        newBlocks = {}
+        for location in uninjectedData:
+            # FIXME TODO REMOVE hack to test RSEs
+            rseName = "%s_Test" % location
+            for container in uninjectedData[location]:
+                for block in uninjectedData[location][container]:
+                    if block not in self.blocksCache:
+                        if self.rucio.createBlock(block, rse=rseName, meta=self.metaData):
+                            logging.info("Block %s inserted into Rucio", block)
+                            newBlocks[block] = location
+                        else:
+                            logging.error("Failed to create block: %s", block)
+        logging.info("Successfully inserted %d blocks into Rucio", newBlocks)
+        return newBlocks
+
+    # TODO: this will likely go away once the phedex to rucio migration is over
+    def _isBlockTierAllowed(self, blockName):
+        """
+        Check the block datatier against the list of data tiers that we want
+        to get pinned in Rucio through a block rule
+        :return: boolean whether we can or not insert a rule
+        """
+        for skipTier in self.skipRulesForTiers:
+            endBlock = blockName.rsplit('/', 1)[1]
+            if endBlock.split('#')[0] == skipTier:
+                return False
+        return True
+
+    def insertBlockRules(self, blocksAndRSEs):
+        """
+        Creates a simple replication rule for every single block that
+        has been previously inserted into Rucio.
+        Right now, we make standard rules against a given site.
+        :param blocksAndRSEs: dictionary with blocks as key, and RSEs as value
+        :return: a set of blocks with successfully created rules
+        """
+        if not self.createBlockRules:
+            return
+
+        # FIXME: this could have been done in bulk, by RSE
+        logging.info("Preparing to create block rules into Rucio...")
+        newBlocks = set()
+        for block, rse in blocksAndRSEs.items():
+            if not self._isBlockTierAllowed(block):
+                logging.info("Component configured to skip block rule for: %s", block)
+                continue
+            # FIXME TODO REMOVE hack to test RSEs
+            rseName = "%s_Test" % rse
+            # DATASET = replicates all files in the same block to the same RSE
+            resp = self.rucio.createReplicationRule(block, rseExpression="rse=%s" % rseName,
+                                                    account=self.rucioAcct, grouping="DATASET",
+                                                    comment="WMCore automatic block rule",
```

Comment:
Can you say something like "WMAgent production site"

Reply:
Good idea!